### PR TITLE
Fix Chef-15 Habitat Builds

### PR DESCRIPTION
**Pull request by christopher-snapp**
## Description

This PR brings the `chef-15` branch into alignment with `master`'s approach to building, testing, and promoting Habitat artifacts.

Signed-off-by: Christopher A. Snapp <csnapp@chef.io>

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [X] I have read the **CONTRIBUTING** document.
- [ ] I have run the pre-merge tests locally and they pass.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [X] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

**Review by lamont-granquist — APPROVED** [review]

**Comment by tas50** [comment]
@christopher-snapp How's this one coming along?

File: habitat/plan.sh [hunk 1]
```diff
@@ -52,10 +55,36 @@ do_verify() {
...
+  ( cd "$CACHE_PATH"
+    gem install bundler -v 1.17.3
+    bundle config --local build.nokogiri "--use-system-libraries \
+        --with-zlib-dir=$(pkg_path_for zlib) \
+        --with-xslt-dir=$(pkg_path_for libxslt) \
+        --with-xml2-include=$(pkg_path_for libxml2)/include/libxml2 \
+        --with-xml2-lib=$(pkg_path_for libxml2)/lib"
+    bundle config --local jobs "$(nproc)"
+    bundle config --local without server docgen maintenance pry travis integration ci chefstyle
+    bundle config --local with development omnibus_package # pull in rake, rspec, + appbundler
```
Comment (christopher-snapp) [inline habitat/plan.sh:83]: this still seems wrong, but without it, there's no rake, rspec, OR appbundler

**Comment by christopher-snapp** [comment]
@tas50 I think this PR is ready for review now that we chased down the last remaining failure by bringing the `executables` tag over from master

**Comment by tas50** [comment]
Congrats. You battled Ruby and won

**Comment by phiggins** [comment]
🎉 💪

**Pull request merged**